cloud-functions/controllers/posts.js

```diff
@@ -0,0 +1,48 @@
+const firebaseDb = require('../firebase-database')
+const utils = require('../utils')
+
+const getAllPosts = function (req, res, next) {
+  var postsRef = firebaseDb.getInstance().collection('posts')
+  postsRef.get()
+    .then(snapshot => {
+      if (snapshot.empty) {
+        return res.status(404).json(utils.getResponseObj(null, 'No posts found', 404))
+      }
+
+      var postList = []
+      snapshot.forEach(doc => {
+        var postData = doc.data()
+        postData.id = doc.id
```

Comment:
burda listeyi donmemize gerek var mi cidden? Simdilik 3-5 post da bisey farketmeyiz ama app de en fazla burasi kullanilacaktir diye tahmin ediyorum. Ne kadar hizli o kadar iyi. bu id yi entry icine storelayalim derim. Mobil taraftaki modellerde de ihtiyacimiz olacak zaten ama modeller icinde id yok henuz.

Reply:
> bu id'yi entry icinde store'layalim
tam olarak ne oluyor?
ben id'yi de body icerisinde donduruyorum ki o post'a tiklanildigi zaman hangi id ile tekrar sorgu yapip gelen datayi yeni bir view'da gosterebilelim.

> burada listeyi donmemize gerek var mi?
buraya bir query parameter sistemi ekleyip belli bir endDate'e kadar sorgu cekebiliriz, daha sonra asagi kaydirdikca yeniden sorgu yapip gelen postlari listenin sonuna append edebiliriz. ne dersiniz @gurolcay @onurccn

Reply:
bir de yukaridaki onerimi farkli bir issue olarak acip baska bir branchte gecirmemiz daha uygun olur. bu issue'nun kapsami degil.

Reply:
> tam olarak ne oluyor?
db ye bi entry ekledigimizde normalde key i neyse (bu normalde id field i olur) onu incremental arttirir ve db girdisi icerisinde id de olur. yani o id deki entry yi istediginde gelen body icinde id de olur. burda kullandigimiz db nin key mantigi falan yok anladigim kadariyla. o yuzden diyorum ki id = 0 olarak insert edelim. insertten sonra doc.id sini alip yeni koydugumuz entry icindeki id yi guncelleyelim. insert u bi tik uzatir ama bu uzatma her insertion islemi icin ayni olur. ama bu sekilde liste get ledigimizde linear olarak uzayacak zaman.

> ben id'yi de body icerisinde donduruyorum ki o post'a tiklanildigi zaman hangi id ile tekrar sorgu yapip gelen datayi yeni bir view'da gosterebilelim.
o id yi de donmemiz lazim zaten ama burda da bak mesela type sikintisina dusuyoruz. o id nin karsiligi yok mesela suan app modelinde.

query onerine katiliyorum.

ikisi icin de ayri issue lar acalim. db modellerine id eklenmesi ve query icin. bu pr kapsaminda da for u kaldir derim o id suan model icerisinde olmayacak zaten.

Reply:
Query: https://trello.com/c/RFlL8WCe
ID in DB Models: https://trello.com/c/oR2x40CS
issuelar acildi.

For'u kaldirmak degil de, return edilen yapinin icinden id'yi kaldirdim:
https://github.com/gtubt/gtubt-mobile/pull/34/commits/c7f6e12c20d6064616a1364f2ce01cbb319d2f8b

Reply:
@YasirNacak flutter tarafinda normalde bu gibi datalari almka icin stream mantigi kullaniliyor. cloudfunction yazarak bunu nasil yapariz arastirir misin? yukaridaki tartismalara son verecek bir sey. once flutter tarafinda stream kullanilarak nasil firebase'den data aliniyor onu bir arastir istersen.